azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsWebMvcAutoConfiguration.java

```diff
@@ -0,0 +1,68 @@
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.boot;
+
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.internal.quickpulse.QuickPulse;
+import com.microsoft.applicationinsights.web.internal.WebRequestTrackingFilter;
+import com.microsoft.applicationinsights.web.spring.internal.InterceptorRegistry;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.Ordered;
+
+@Configuration
+@Import(InterceptorRegistry.class)
+@ConditionalOnBean(TelemetryConfiguration.class)
+@ConditionalOnWebApplication
+@AutoConfigureAfter(ApplicationInsightsTelemetryAutoConfiguration.class)
+public class ApplicationInsightsWebMvcAutoConfiguration {
+
+    @Bean
+    public FilterRegistrationBean webRequestTrackingFilterRegistrationBean(WebRequestTrackingFilter webRequestTrackingFilter) {
+        FilterRegistrationBean registration = new FilterRegistrationBean();
+        registration.setFilter(webRequestTrackingFilter);
+        registration.addUrlPatterns("/*");
+        registration.setName("webRequestTrackingFilter");
+        registration.setOrder(Ordered.HIGHEST_PRECEDENCE + 10);
+        return registration;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public WebRequestTrackingFilter webRequestTrackingFilter(@Value("${spring.application.name:application}") String applicationName) {
+        return new WebRequestTrackingFilter(applicationName);
+    }
+
+    @Bean
```

Comment:
This should also be conditional as far as I know. There should be a way to disable quick pulse if needed. In Standard xml implementation we have QuickPulse XML element which can be set to disabled. We need to have that property here too and then se this bean on conditional with that.

Reply:
Oh, probably I missed this one, actually I had a way to disable quick pulse but then I removed it because it seemed to me useless thing, I will return it back.

Reply:
Yes please do add. There should be ability to disable quick pulse if needed.

Reply:
@dhaval24 should quick pulse depend on web application or it's applies to non web as well?

Reply:
@gavlyukovskiy frankly, I do not know about this on the top of my head. I haven't worked on this part of the code base (QuickPulse), however from an end-user perspective QuickPulse / Live View on portal shows information such as Incoming request rate, request duration, request failure rate, CPU usage, Memory usage, Exception Rate, Dependency rate, dependency failure rate and Dependency call rate. Based on this it seems like QuickPulse should not have explicit dependency on web application. Only web related features like Request parameters should not be populated on the portal.